plugin/src/App/init.lua

```diff
@@ -59,15 +63,57 @@ function App:startSession(host, port, sessionOptions)
 				appStatus = AppStatus.Connecting,
 			})
 		elseif status == ServeSession.Status.Connected then
+			if expectedSessionId and details.sessionId ~= expectedSessionId then
+				ServeSession:stop()
+			end
+
 			local address = ("%s:%s"):format(host, port)
+
+			if RunService:IsClient() and RunService:IsServer() and not RunService:IsRunning() then
+				local connectionId = HttpService:GenerateGUID()
+				local activeConnections = settings:get("activeConnections", true)
+				table.insert(activeConnections, {
+					connectionId = connectionId,
+					sessionId = details.sessionId,
+					host = host,
+					port = port
+				})
+
+				if #activeConnections > 15 then
+					table.remove(activeConnections, 1)
+				end
+
+				settings:set("activeConnections", activeConnections)
+
+				workspace:SetAttribute("__RojoConnectionId", connectionId)
```

Comment:
This is putting temporary data into a permanent location- could we use a non-Archivable Instance for this, or do they disappear when you press Play Solo? It might simplify the cleanup logic.